decorator.rb

```diff
@@ -0,0 +1,26 @@
+require './nameable'
+
+class BaseDecorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.strip.upcase
+  end
```

Comment:
- [ ] When you use the `upcase` method, the whole string is converted to uppercase while the `capitalize` method converts the first letter in the string to uppercase while others are left as they are. Kindly use capitalize based on the project's requirement. 👍